gempy/core/interpolator.py

```diff
@@ -138,20 +148,56 @@ def set_theano_shared_kriging(self):
             True
         """
         # Range
+        range_val = self.additional_data.kriging_data.df.loc['values', 'range']
+        range_res = range_val / self.additional_data.rescaling_data.df.loc[
+            'values', 'rescaling factor']
+
+        range_list = range_res * np.ones(
+            self.additional_data.structure_data.df.loc['values',
+                                                       'number series'])
+        #
+        # if type(range_res) is np.float or type(range_res) is np.float64:
+        #     range_list = range_res * np.ones(
+        #         self.additional_data.structure_data.df.loc['values',
+        #                                                    'number series']
+        #     )
+        # elif isinstance(range_res, Iterable):
+        #     range_list = range_res
+        # else:
+        #     raise AttributeError('Range must be either int or Iterable')
+
         # TODO add rescaled range and co into the rescaling data df?
-        self.theano_graph.a_T.set_value(np.cast[self.dtype]
-                                        (self.additional_data.kriging_data.df.loc['values', 'range'] /
-                                         self.additional_data.rescaling_data.df.loc[
-                                             'values', 'rescaling factor']))
+        self.theano_graph.a_T.set_value(np.cast[self.dtype](range_list))
+
         # Covariance at 0
-        self.theano_graph.c_o_T.set_value(np.cast[self.dtype](
-            self.additional_data.kriging_data.df.loc['values', '$C_o$'] /
-            self.additional_data.rescaling_data.df.loc[
-                'values', 'rescaling factor']
-        ))
+
+        cov_val = self.additional_data.kriging_data.df.loc['values', '$C_o$']
+        cov_res = cov_val / self.additional_data.rescaling_data.df.loc[
+            'values', 'rescaling factor']
+        cov_list = cov_res * np.ones(
+            self.additional_data.structure_data.df.loc['values',
+                                                       'number series']
+        )
+        #
+        #
+        # if type(cov_res) is np.float or type(cov_res) is np.float64:
+        #     cov_list = cov_res * np.ones(
+        #         self.additional_data.structure_data.df.loc['values',
+        #                                                    'number series']
+        #     )
+        # elif isinstance(cov_res, Iterable):
+        #     cov_list = cov_res
+        # else:
+        #     raise AttributeError('Covariance at 0 must be either int or Iterable')
+
+        self.theano_graph.c_o_T.set_value(
+            np.cast[self.dtype](cov_list)
+        )
+
         # universal grades
         self.theano_graph.n_universal_eq_T.set_value(
-            list(self.additional_data.kriging_data.df.loc['values', 'drift equations'].astype('int32')[self.non_zero]))
+            list(self.additional_data.kriging_data.df.loc['values', 'drift equations'].astype(
```

Comment:
I think the code could be a lot cleaner to read if getter methods were used instead of direct reference to the underlying dataframes

Reply:
Agree.In the next months I am going to revise the data objects and part of the revision will be to split columns between public and private and for the public create properties